backend/api/routes/files.py

```diff
@@ -83,15 +84,25 @@ def generate_file_hash(file: BinaryIO) -> str:
     return hasher.hexdigest()
 
 
-def upload_file(location: Path, file: BinaryIO) -> Path:
+def save_file(file_name: str, file: BinaryIO) -> Path:
```

Comment:
I think it makes more sense to have the file (main subject) be the first arg
```suggestion
def save_file(file: BinaryIO, file_name: str) -> Path:
```